main/streams/glob_wrapper.c

```diff
@@ -224,6 +235,19 @@ static php_stream *php_glob_stream_opener(php_stream_wrapper *wrapper, const cha
 		}
 	}
 
+	/* if open_basedir in use, check and filter restricted paths */
+	if ((options & STREAM_DISABLE_OPEN_BASEDIR) == 0) {
+		pglob->open_basedir_indexmap_used = true;
+		for (i = 0; i < pglob->glob.gl_pathc; i++) {
+			if (!php_check_open_basedir_ex(pglob->glob.gl_pathv[i], 0)) {
+				if (!pglob->open_basedir_indexmap) {
+					pglob->open_basedir_indexmap = (size_t *) emalloc(sizeof(size_t) * pglob->glob.gl_pathc);
```

Comment:
It's highly unlikely that this overflows, but we could make sure:
```suggestion
					pglob->open_basedir_indexmap = (size_t *) safe_emalloc(pglob->glob.gl_pathc, sizeof(size_t), 0);
```